aten/src/ATen/cudnn/Descriptors.h

```diff
@@ -148,10 +186,105 @@ struct SpatialTransformerDescriptor
                       &cudnnCreateSpatialTransformerDescriptor,
                       &cudnnDestroySpatialTransformerDescriptor>
 {
-  SpatialTransformerDescriptor() : Descriptor() {
-  }
   void set(cudnnDataType_t dataType, int dim, int* size) {
-    CUDNN_CHECK(cudnnSetSpatialTransformerNdDescriptor(desc(), CUDNN_SAMPLER_BILINEAR, dataType, dim, size));
+    CUDNN_CHECK(cudnnSetSpatialTransformerNdDescriptor(mut_desc(), CUDNN_SAMPLER_BILINEAR, dataType, dim, size));
+  }
+};
+
+#if CUDNN_VERSION < 7000
+
+inline cudnnStatus_t cudnnRestoreDropoutDescriptor(
+    cudnnDropoutDescriptor_t dropoutDesc,
+    cudnnHandle_t handle,
+    float dropout,
+    void *states,
+    size_t stateSizeInBytes,
+    unsigned long long seed) {
+  // Try to accurately simulate cuDNN's behavior, for our cuDNN 6 friends
+  if (states == nullptr) return CUDNN_STATUS_INVALID_VALUE;
+  if (stateSizeInBytes == 0) return CUDNN_STATUS_INVALID_VALUE;
+  dropoutDesc->dropout = dropout;
+  dropoutDesc->nstates = stateSizeInBytes;
```

Comment:
No, he did not, this is not actually right ;-)

Reply:
Nooooooooo :o)

(What actually happened is that I thought the function permitted certain types of arguments, but when I ran it in CUDA 9, it did not, so I added some checks which I thought might catch these sorts of errors if you were developing on CUDA 8. Happy to improve this however necessary, but this seemed OK for hack :)